pkg/kubelet/dockertools/docker_test.go

```diff
@@ -327,3 +327,95 @@ func TestIsImagePresent(t *testing.T) {
 		t.Errorf("expected inspection of image abc:123, instead inspected image %v", cl.imageName)
 	}
 }
+
+func TestGetRunningContainers(t *testing.T) {
+	fakeDocker := &FakeDockerClient{}
+	tests := []struct {
+		containers  map[string]*docker.Container
+		inputIDs    []string
+		expectedIDs []string
+		err         error
+	}{
+		{
+			containers: map[string]*docker.Container{
+				"foobar": {
+					State: docker.State{
+						Running: false,
+					},
+				},
+				"baz": {
+					State: docker.State{
+						Running: true,
+					},
+				},
+			},
+			inputIDs:    []string{"foobar", "baz"},
+			expectedIDs: []string{"baz"},
+		},
+		{
+			containers: map[string]*docker.Container{
+				"foobar": {
+					State: docker.State{
+						Running: true,
+					},
+				},
+				"baz": {
+					State: docker.State{
+						Running: true,
+					},
+				},
+			},
+			inputIDs:    []string{"foobar", "baz"},
+			expectedIDs: []string{"foobar", "baz"},
+		},
+		{
+			containers: map[string]*docker.Container{
+				"foobar": {
+					State: docker.State{
+						Running: false,
+					},
+				},
+				"baz": {
+					State: docker.State{
+						Running: false,
+					},
+				},
+			},
+			inputIDs:    []string{"foobar", "baz"},
+			expectedIDs: []string{},
+		},
+		{
+			containers: map[string]*docker.Container{
+				"foobar": {
+					State: docker.State{
+						Running: false,
+					},
+				},
+				"baz": {
+					State: docker.State{
+						Running: false,
+					},
+				},
+			},
+			inputIDs: []string{"foobar", "baz"},
+			err:      fmt.Errorf("test error"),
+		},
+	}
+	for _, test := range tests {
+		fakeDocker.ContainerMap = test.containers
+		fakeDocker.Err = test.err
+		results, err := GetRunningContainers(fakeDocker, test.inputIDs)
```

Comment:
My usual super minor style comment here: you can move the assignment into the if.